tests/ui/unnecessary_lazy_eval.rs

```diff
@@ -2,6 +2,7 @@
 #![warn(clippy::unnecessary_lazy_evaluations)]
 #![allow(clippy::redundant_closure)]
 #![allow(clippy::bind_instead_of_map)]
```

Comment:
Is this still necessary with the latest changes?

Reply:
`#![allow(clippy::bind_instead_of_map)]` ?
yes otherwise it lints the blocks commented with `// should not lint, bind_instead_of_map takes priority`

Reply:
> For future contributions, it would be great if you could submit multiple commits for changes this big and then squash them right before merging. This makes reviewing way easier.

Sure, will do.

Reply:
> yes otherwise it lints the blocks commented with `// should not lint, bind_instead_of_map takes priority`

Ah of course!

> Sure, will do.

Thanks! For large contributions like this, squashes aren't even necessary, if commits (like "Update \<filename\>") with one-line changes are prevented from the beginning.